### DISCUSSION TITLE
How to push local development seed data to production

### DISCUSSION THREAD

**Discussion by SEI-John** [supabase/supabase#26393] [Questions]
I'm wondering how to sync seed data for my production environment. 

I've had no problem setting up a local instance of supabase with `supabase init`, logging in, then pulling from the remote db. I brought in seed data from the production server through `supabase db dump --data-only -f supabase/seed.sql` and imported it through `psql -h localhost -U postgres -p 54322 -f supabase/seed.sql` easily too.

Now, I've done some development on my local machine and want to push some seed data up to my production server. Is this possible? I don't see many tutorials on how to go the other direction. 

Edit: I've been able to sync the schema changes through `supabase db push`, it's just the seed data I want to push up.

**Comment by ThingEngineer** [comment]
As you probably know there isn't a cli command to push seed data to your remote db. In cases where I need to do this I just copy the seed data that I want to apply to the remote db from the local seed.sql file or custom dump, then paste it into the Supabase Studio SQL Editor. 

Since this is more of a one off or at least rare task, and it can also be very dangerous, there will probably never be a command to just push data to your live remote db.

A possible option if this is very important or a frequent task for you are tools like Flyway or Liquibase for database migrations and seeding. These tools can connect to your Supabase database (PostgreSQL) and manage schema changes and seed data deployments.

**Reply by tillka** [reply]
> As you probably know there isn't a cli command to push seed data to your remote db. In cases where I need to do this I just copy the seed data that I want to apply to the remote db from the local seed.sql file or custom dump, then paste it into the Supabase Studio SQL Editor.
> 
> Since this is more of a one off or at least rare task, and it can also be very dangerous, there will probably never be a command to just push data to your live remote db.
> 
> A possible option if this is very important or a frequent task for you are tools like Flyway or Liquibase for database migrations and seeding. These tools can connect to your Supabase database (PostgreSQL) and manage schema changes and seed data deployments.

Maybe I understood the question or your comment wrong, but isn't `supabase db push  --include-seed` exactly doing this?

**Reply by jpteixeira99** [reply]
> > As you probably know there isn't a cli command to push seed data to your remote db. In cases where I need to do this I just copy the seed data that I want to apply to the remote db from the local seed.sql file or custom dump, then paste it into the Supabase Studio SQL Editor.
> > Since this is more of a one off or at least rare task, and it can also be very dangerous, there will probably never be a command to just push data to your live remote db.
> > A possible option if this is very important or a frequent task for you are tools like Flyway or Liquibase for database migrations and seeding. These tools can connect to your Supabase database (PostgreSQL) and manage schema changes and seed data deployments.
> 
> Maybe I understood the question or your comment wrong, but isn't `supabase db push --include-seed` exactly doing this?

I'm trying to leverage this to push local data to my remote staging project. I've made sure that `npx supabase link` is pointing to it, generated the `seed.sql` file with `npx supabase db dump --data-only --schema "public,storage" --local -f "seed.sql"` and placed it under `supabase/seed.sql` but when I run `npx supabase db push --include-seed` it just returns "Remote database is up to date" but doesn't actually update any data...

**Reply by ThingEngineer** [reply]
> Maybe I understood the question or your comment wrong, but isn't `supabase db push --include-seed` exactly doing this?

I stand corrected. That is pretty new and I did not notice it. Also it looks like it may still have issues. Not sure what issue @Viriatto is facing exactly without testing it myself. 

Quite new apparently: #13832
